src/gwt/src/org/rstudio/studio/client/workbench/views/files/FilesUpload.java

```diff
@@ -67,22 +67,27 @@ public void execute()
          new OperationWithInput<PendingFileUpload>() {
             public void execute(PendingFileUpload pendingUpload)
             {
-               // confirm overwrites if necessary
+               Boolean unzipFound = pendingUpload.getUnzipFound();
+               Boolean isZip = pendingUpload.getIsZip();
                FileUploadToken token = pendingUpload.getToken();
```

Comment:
I could see how to get values out of pendingUpload, but not from token. So unzipFound is in both. Maybe not a big deal, but if there's a cleaner way to do it I'm all ears.

Reply:
This is fine to me as well, but I think you could just access it from the FileUploadToken in both cases.